src/sdo/sood/data/sdo_ml_v2_dataset.py

```diff
@@ -119,6 +122,28 @@ def __init__(
             attrs = data.attrs
             all_images = da.from_array(data)
 
+        if irradiance is not None:
```

Comment:
this is the important part ^^ We will still need to check if this is efficient when being run on the full dataset